cosign isn't installed properly if AQUA_GOOS and AQUA_GOARCH are set

## aqua version

Please use the latest version.

```console
$ aqua -v
aqua version 1.30.0 (437c21526e716097dd8622a05dc0a12b068fb238)
```

## Environment

* OS (Windows, Linux, macOS, etc): linux
* CPU Architecture (amd64, arm64, etc): amd64
* AQUA_GOOS: darwin
* AQUA_GOARCH: arm64

## Overview

* https://github.com/aquaproj/aqua-registry/actions/runs/3846994526/jobs/6552935769
* https://github.com/aquaproj/aqua-registry/pull/8911

```
INFO[0001] download and unarchive the package            aqua_version=1.30.0 env=darwin/arm64 package_name=sigstore/cosign package_version=v1.13.1 program=aqua registry=
ERRO[0007] install the package                           aqua_version=1.30.0 asset_name=registry-tool_darwin_arm64.tar.gz env=darwin/arm64 error="verify a checksum file with Cosign: verify a signature file with Cosign: verify with cosign: fork/exec /home/runner/.local/share/aquaproj-aqua/pkgs/github_release/github.com/sigstore/cosign/v1.13.1/cosign-linux-amd64/cosign-linux-amd64: no such file or directory" package_name=aquaproj/registry-tool package_version=v0.1.6 program=aqua registry=standard
FATA[0007] aqua failed                                   aqua_version=1.30.0 env=darwin/arm64 error="it failed to install some packages" program=aqua
Error: Process completed with exit code 1.
```

```
verify a checksum file with Cosign:
verify a signature file with Cosign:
verify with cosign:
fork/exec /home/runner/.local/share/aquaproj-aqua/pkgs/github_release/github.com/sigstore/cosign/v1.13.1/cosign-linux-amd64/cosign-linux-amd64:
no such file or directory
```

## How to reproduce

* aqua.yaml
* executed command and output

> aqua.yaml should be not partial but complete configuration.
> Please remove unneeded configuration to reproduce the issue.

```yaml
# aqua.yaml

```

```console
$ 
```

## Debug Output

Please set the environment variable `AQUA_LOG_LEVEL=debug` and run aqua and tell us the debug output.

```console
$ 
```

## Expected Behaviour

## Actual Behaviour

## Important Factoids

## References

*

--- a/pkg/installpackage/checksum.go
+++ b/pkg/installpackage/checksum.go
@@ -59,7 +59,7 @@ func (inst *InstallerImpl) dlAndExtractChecksum(ctx context.Context, logE *logru
 		}
 		art := pkg.GetTemplateArtifact(inst.runtime, assetName)
 		logE.Info("verify a checksum file with Cosign")
-		if err := inst.installCosign(ctx, logE, cosign.Version); err != nil {
+		if err := inst.cosignInstaller.installCosign(ctx, logE, cosign.Version); err != nil {
 			return "", err
 		}
 		if err := inst.cosign.Verify(ctx, logE, inst.runtime, &download.File{

--- a/pkg/installpackage/cosign.go
+++ b/pkg/installpackage/cosign.go
@@ -12,88 +12,11 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// type Cosign struct {
-// 	installer *InstallerImpl
-// }
-//
-// func NewCosign(param *config.Param, downloader download.ClientAPI, fs afero.Fs, linker domain.Linker, executor Executor, chkDL download.ChecksumDownloader, chkCalc ChecksumCalculator, unarchiver Unarchiver, policyChecker policy.Checker, cosignVerifier cosign.Verifier, slsaVerifier slsa.Verifier) *Cosign {
-// 	return &Cosign{
-// 		installer: &InstallerImpl{
-// 			rootDir:            param.RootDir,
-// 			maxParallelism:     param.MaxParallelism,
-// 			downloader:         downloader,
-// 			checksumDownloader: chkDL,
-// 			checksumFileParser: &checksum.FileParser{},
-// 			checksumCalculator: chkCalc,
-// 			runtime:            runtime.NewR(),
-// 			fs:                 fs,
-// 			linker:             linker,
-// 			executor:           executor,
-// 			progressBar:        param.ProgressBar,
-// 			isTest:             param.IsTest,
-// 			onlyLink:           param.OnlyLink,
-// 			copyDir:            param.Dest,
-// 			unarchiver:         unarchiver,
-// 			policyChecker:      policyChecker,
-// 			cosign:             cosignVerifier,
-// 			slsaVerifier:       slsaVerifier,
-// 		},
-// 	}
-// }
-//
-// func (cos *Cosign) InstallCosign(ctx context.Context, logE *logrus.Entry, version string) error {
-// 	assetTemplate := `cosign-{{.OS}}-{{.Arch}}`
-// 	pkg := &config.Package{
-// 		Package: &aqua.Package{
-// 			Name:    "sigstore/cosign",
-// 			Version: version,
-// 		},
-// 		PackageInfo: &registry.PackageInfo{
-// 			Type:      "github_release",
-// 			RepoOwner: "sigstore",
-// 			RepoName:  "cosign",
-// 			Asset:     &assetTemplate,
-// 			SupportedEnvs: []string{
-// 				"darwin",
-// 				"linux",
-// 				"amd64",
-// 			},
-// 		},
-// 	}
-//
-// 	chksum := cosign.Checksums()[cos.installer.runtime.Env()]
-//
-// 	pkgInfo, err := pkg.PackageInfo.Override(pkg.Package.Version, cos.installer.runtime)
-// 	if err != nil {
-// 		return fmt.Errorf("evaluate version constraints: %w", err)
-// 	}
-// 	supported, err := pkgInfo.CheckSupported(cos.installer.runtime, cos.installer.runtime.GOOS+"/"+cos.installer.runtime.GOARCH)
-// 	if err != nil {
-// 		return fmt.Errorf("check if cosign is supported: %w", err)
-// 	}
-// 	if !supported {
-// 		logE.Debug("the package isn't supported on this environment")
-// 		return nil
-// 	}
-//
-// 	pkg.PackageInfo = pkgInfo
-//
-// 	if err := cos.installer.InstallPackage(ctx, logE, &ParamInstallPackage{
-// 		Checksums: checksum.New(), // Check cosign's checksum but not update aqua-checksums.json
-// 		Pkg:       pkg,
-// 		Checksum: &checksum.Checksum{
-// 			Algorithm: "sha256",
-// 			Checksum:  chksum,
-// 		},
-// 		// PolicyConfigs is nil, so the policy check is skipped
-// 	}); err != nil {
-// 		return err
-// 	}
-//
-// 	return nil
-// }
+type Cosign struct {
+	installer *InstallerImpl
+}
 
-func (inst *InstallerImpl) installCosign(ctx context.Context, logE *logrus.Entry, version string) error {
+func (cos *Cosign) installCosign(ctx context.Context, logE *logrus.Entry, version string) error {
 	assetTemplate := `cosign-{{.OS}}-{{.Arch}}`
 	pkg := &config.Package{
 		Package: &aqua.Package{
@@ -113,13 +36,13 @@ func (inst *InstallerImpl) installCosign(ctx context.Context, logE *logrus.Entry
 		},
 	}
 
-	chksum := cosign.Checksums()[inst.runtime.Env()]
+	chksum := cosign.Checksums()[cos.installer.runtime.Env()]
 
-	pkgInfo, err := pkg.PackageInfo.Override(pkg.Package.Version, inst.runtime)
+	pkgInfo, err := pkg.PackageInfo.Override(pkg.Package.Version, cos.installer.runtime)
 	if err != nil {
 		return fmt.Errorf("evaluate version constraints: %w", err)
 	}
-	supported, err := pkgInfo.CheckSupported(inst.runtime, inst.runtime.Env())
+	supported, err := pkgInfo.CheckSupported(cos.installer.runtime, cos.installer.runtime.Env())
 	if err != nil {
 		return fmt.Errorf("check if cosign is supported: %w", err)
 	}
@@ -130,7 +53,7 @@ func (inst *InstallerImpl) installCosign(ctx context.Context, logE *logrus.Entry
 
 	pkg.PackageInfo = pkgInfo
 
-	if err := inst.InstallPackage(ctx, logE, &ParamInstallPackage{
+	if err := cos.installer.InstallPackage(ctx, logE, &ParamInstallPackage{
 		Checksums: checksum.New(), // Check cosign's checksum but not update aqua-checksums.json
 		Pkg:       pkg,
 		Checksum: &checksum.Checksum{

--- a/pkg/installpackage/download.go
+++ b/pkg/installpackage/download.go
@@ -98,7 +98,7 @@ func (inst *InstallerImpl) download(ctx context.Context, logE *logrus.Entry, par
 	if cos := ppkg.PackageInfo.Cosign; cos.GetEnabled() {
 		art := ppkg.GetTemplateArtifact(inst.runtime, param.Asset)
 		logE.Info("verify a package with Cosign")
-		if err := inst.installCosign(ctx, logE, cosign.Version); err != nil {
+		if err := inst.cosignInstaller.installCosign(ctx, logE, cosign.Version); err != nil {
 			return err
 		}
 		if err := inst.cosign.Verify(ctx, logE, inst.runtime, &download.File{

--- a/pkg/installpackage/installer.go
+++ b/pkg/installpackage/installer.go
@@ -46,9 +46,18 @@ type InstallerImpl struct {
 	isTest             bool
 	copyDir            string
 	policyChecker      policy.Checker
+	cosignInstaller    *Cosign
 }
 
 func New(param *config.Param, downloader download.ClientAPI, rt *runtime.Runtime, fs afero.Fs, linker domain.Linker, executor Executor, chkDL download.ChecksumDownloader, chkCalc ChecksumCalculator, unarchiver Unarchiver, policyChecker policy.Checker, cosignVerifier cosign.Verifier, slsaVerifier slsa.Verifier) *InstallerImpl {
+	installer := newInstaller(param, downloader, rt, fs, linker, executor, chkDL, chkCalc, unarchiver, policyChecker, cosignVerifier, slsaVerifier)
+	installer.cosignInstaller = &Cosign{
+		installer: newInstaller(param, downloader, runtime.NewR(), fs, linker, executor, chkDL, chkCalc, unarchiver, policyChecker, cosignVerifier, slsaVerifier),
+	}
+	return installer
+}
+
+func newInstaller(param *config.Param, downloader download.ClientAPI, rt *runtime.Runtime, fs afero.Fs, linker domain.Linker, executor Executor, chkDL download.ChecksumDownloader, chkCalc ChecksumCalculator, unarchiver Unarchiver, policyChecker policy.Checker, cosignVerifier cosign.Verifier, slsaVerifier slsa.Verifier) *InstallerImpl {
 	return &InstallerImpl{
 		rootDir:            param.RootDir,
 		maxParallelism:     param.MaxParallelism,

